LSIF: Docmentation updates (#6898)
sourcegraph/sourcegraph: doc/user/code_intelligence/lsif_quickstart.md

$ src \
  -endpoint=https://sourcegraph.example.com \
  lsif upload \
  -file=<LSIF file (e.g. ./cmd/dump.lsif)>
```

> If uploading to sourcegraph.com, you will need to additionally supply the `-github-token=<token>` argument. This token should have repo read access to your user account and ensures that you have collaborator access to the repository for which you are uploading data.

Reviewer: ```suggestion
If uploading to Sourcegraph.com, you will need to additionally supply the `-github-token=<token>` argument. This token must have the `repo` scope. It is used to verify that you have collaborator access to the repository for which you are uploading data.
```

- Capitalize Sourcegraph.com (per https://about.sourcegraph.com/handbook/communication/style_guide#referring-to-the-product-and-features)
- `must` not `should`
- use the precise terms for the scope required
